src/java.base/share/classes/java/lang/constant/DirectMethodHandleDesc.java

```diff
@@ -137,13 +137,14 @@ public static Kind valueOf(int refKind, boolean isInterface) {
             if (i >= 0 && i < TABLE.length) {
                 Kind kind = TABLE[i];
                 if (kind == null) {
-                    throw new IllegalArgumentException(String.format("refKind=%d", refKind));
+                    throw new IllegalArgumentException(String.format("refKind=%d isInterface=%s", refKind, isInterface));
                 }
-                if (kind.refKind == refKind && kind.isInterface == isInterface) {
+                if (kind.refKind == refKind &&
+                        (refKind != REF_invokeStatic || refKind != REF_invokeSpecial || kind.isInterface == isInterface)){
```

Comment:
It reads to me that the static initializer tries to set up the table such that `valueOf(refKind, isInterface)` should find the proper kind to return except this:
```
// There is not a specific Kind for interfaces
if (kind == VIRTUAL) kind = INTERFACE_VIRTUAL;
```
This changes the entry for `REF_invokeVirtual` to kind `INTERFACE_VIRTUAL`.  Do you know why?    If this entry is set to VIRTUAL, then each refKind has two entries in the table corresponding to the correct result for `valueOf`.